todo-app/src/stores/TaskStore.js

```diff
@@ -0,0 +1,52 @@
+import {autorun, observable, action} from 'mobx';
+import {loadTodoItems, saveTodoItems} from '../storage'
+
+
+export default class TaskStore {
+  constructor() {
+    autorun(() => {
+      saveTodoItems(this.todos);
+    });
+  }
+    
+    @observable
+    todos = loadTodoItems();
+    
+
+    @action
+    addTask(text){
+        const ids = this.todos.map(task => task.id);
+        const todo = {
+            id: ids.length === 0 ? 1 : Math.max(...ids) + 1,
+            description: text,
+            done: false
+        }
+       this.todos.push(todo);  
+    }
+    @action
+      removeTask(id) {
+      this.todos = this.todos.filter(task => task.id !== id)
+  }
+     @action
+     toggleStatus(todoID) {
```

Comment:
What is going on in this function? You are mapping through `this.todos` on your store and assigning that new array to a value of `todo`, and then returning `todo` _inside_ your map block, which is undefined.

So your `const todo` ends up equaling `[undefined, undefined, undefined]`.

It appears it's completely an accident that this function works. You are mutating the specific value directly inside your `map` function.

You have two options:

1. Mutate the specific todo directly without doing this unnecessary `map` and assigning an array of `undefined`s to a `const`. This would look like:

>const todoToToggle = this.todos.filter(todo => todo.id === todoID)[0]
> // Now reverse the value

2. Return a new array properly, without mutating your old data, and set `this.todos` to it.

I _highly_ recommend attempting the second option. It seems like you already had this in mind, you just lost track halfway through.

The first clue is that `todo` is of course undefined inside your `map` block.